### PR TITLE
Dataloading should fail if requested Mixture is not feasible

### PR DESCRIPTION
**Pull request by ahmeda14960**
Right now if we request mixtures through LMMixtureDataset that have neither the raw urls or a finished cached, levanter will just skip and not include the dataset in the mixture. This can lead to bad silent errors so instead we will complain loudly and fail

### CODE REVIEW

File: src/levanter/data/text.py [hunk 1]
```diff
@@ -1093,8 +1093,14 @@ def build_or_load_cache(
                 pass
 
         if source is None:
-            logger.warning(f"Skipping {split} because no source was provided")
-            return None
+            if split == "train":
```
Comment (rjpower) [inline src/levanter/data/text.py:1096]: Why is "train" special, should we always fail if we don't have the source?

Comment (dlwh) [reply src/levanter/data/text.py:1096]: ok yeah so this isn't right. we have lots of sources where there is no training set (e.g. paloma)

Comment (ahmeda14960) [reply src/levanter/data/text.py:1096]: Oh I was thinking we might not have a validation set while training but still continue training.... for example we might not have validation urls for each source in common pile, in which case this would fail.

For paloma, we always use it with a data config that has training urls correct? so this seems fine.

What are you thinking instead? require validation for each train source?
